### ISSUE TITLE
trying to install archzfs on archarm : `404`  on `pacman -Sy` 

### ISSUE THREAD

**Issue by vmonteco** [archzfs/archzfs#392]
I'm on a raspberry pi 2 (armv7) with archlinux arm installed.
I tried the instructions from the wiki (updating the `pacman.conf`  file, setting the keys, updating...)
But when I try to `sudo pacman -Sy`, I get the following : 

```
 sudo pacman -Sy         
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 alarm is up to date
 aur is up to date
error: failed retrieving file 'archzfs.db' from archzfs.com : The requested URL returned error: 404
error: failed to update archzfs (failed to retrieve some files)
error: failed to synchronize all databases
```

I note that it's successfully done on my laptop under archlinux. But I'd like to experiment with it on my rpi.

Is such installation possible? If no, a build from source perhaps?

Regards

**Comment by dmshimself** [comment]
I don't think you can install ZFS on the 32 bit OS that I think you will be using with a PI2.  Pi3 or 4 for a 64 bit OS.  I installed arch for arm, 64 bit on a Pi4 with 4G of memry and ZFS built fine and ran.  I'm not sure how well it is supported at that platform, but it worked quite nicely on that rig.

**Comment by vmonteco** [comment]
I see. Thank you for that information.

Do you know what kind of limitation or requirement would make a build difficult on 32bits architectures?
Are there some specific steps I could try to make a build?

**Comment by dmshimself** [comment]
Well ZFS was written for a 64 bit archtecture and so you are bound to run into problems with a 32 bit one as that isn't what the devlopers use.  Mind you, they didn't use the arm architecture eiher :-)  Also with a max of 1G of memory on the Pi2 you will likely get quite poor performance.  I used a Pi4 with 4G and arch64 install as described here:

https://archlinuxarm.org/platforms/armv8/broadcom/raspberry-pi-4

Note the comments about initialising the pacman keyring.

I then installed yay from the AUR and used that to install zfs-utils and zfs-dkms from AUR.  I did get a warning that the package wan't built for the arm archtitecture, but it built fine and allowed me to import an existing ZFS dataset and also create new ones on some thumbdrives.

Without EEC memory on the Pi, there is a reasonable risk you could get data corruption without detection.
